javagen/src/main/java/com/azure/autorest/mapper/ProxyMethodMapper.java

```diff
@@ -193,6 +193,10 @@ protected void buildUnexpectedResponseExceptionTypes(ProxyMethod.Builder builder
         }
 
         if (errorType != null) {
+            if (errorType.equals(ClassType.String)){
+                builder.unexpectedResponseExceptionType(getExceptionTypeForString(errorType));
+                return;
+            }
```

Comment:
What makes android different from Java which requires special handling for string type?

Reply:
On Android we'd throw this exception.

> com.azure.android.core.http.exception.HttpResponseException

For normal Java, it'll throw

> com.azure.core.exception.HttpResponseException